File: sdks/typescript/src/models/summarization.ts
/**
 * Summarization models for the Hive Vectorizer TypeScript SDK.
 * 
 * This module contains all the data models used for text and context summarization.
 */

export interface SummarizeTextRequest {
  /** Text to summarize */
  text: string;
  /** Summarization method (extractive, keyword, sentence, abstractive) */
  method?: string;
  /** Maximum summary length (optional) */
  max_length?: number;
  /** Compression ratio (optional) */
  compression_ratio?: number;
  /** Language code (optional) */
  language?: string;
  /** Additional metadata (optional) */
  metadata?: Record<string, string>;
}

export interface SummarizeTextResponse {
  /** Summary ID */
  summary_id: string;
  /** Original text */
  original_text: string;
  /** Generated summary */
  summary: string;
  /** Method used */
  method: string;
  /** Original text length */
  original_length: number;
  /** Summary length */
  summary_length: number;
  /** Compression ratio */
  compression_ratio: number;
  /** Language */
  language: string;
  /** Status */
  status: string;
  /** Message */
  message: string;
  /** Metadata */
  metadata: Record<string, string>;
}

export interface SummarizeContextRequest {
  /** Context to summarize */
  context: string;
  /** Summarization method (extractive, keyword, sentence, abstractive) */
  method?: string;
  /** Maximum summary length (optional) */
  max_length?: number;
  /** Compression ratio (optional) */
  compression_ratio?: number;
  /** Language code (optional) */
  language?: string;
  /** Additional metadata (optional) */
  metadata?: Record<string, string>;
}

export interface SummarizeContextResponse {
  /** Summary ID */
  summary_id: string;
  /** Original context */
  original_context: string;
  /** Generated summary */
  summary: string;
  /** Method used */
  method: string;
  /** Original context length */
  original_length: number;
  /** Summary length */
  summary_length: number;
  /** Compression ratio */
  compression_ratio: number;
  /** Language */
  language: string;
  /** Status */
  status: string;
  /** Message */
  message: string;
  /** Metadata */
  metadata: Record<string, string>;
}

export interface GetSummaryResponse {
  /** Summary ID */
  summary_id: string;
  /** Original text */
  original_text: string;
  /** Generated summary */
  summary: string;
  /** Method used */
  method: string;
  /** Original text length */
  original_length: number;
  /** Summary length */
  summary_length: number;
  /** Compression ratio */
  compression_ratio: number;
  /** Language */
  language: string;
  /** Creation timestamp */
  created_at: string;
  /** Metadata */
  metadata: Record<string, string>;
  /** Status */
  status: string;
}

export interface SummaryInfo {
  /** Summary ID */
  summary_id: string;
  /** Method used */
  method: string;
  /** Language */
  language: string;
  /** Original text length */
  original_length: number;
  /** Summary length */
  summary_length: number;
  /** Compression ratio */
  compression_ratio: number;
  /** Creation timestamp */
  created_at: string;
  /** Metadata */
  metadata: Record<string, string>;
}

export interface ListSummariesResponse {
  /** List of summaries */
  summaries: SummaryInfo[];
  /** Total count */
  total_count: number;
  /** Status */
  status: string;
}

export interface ListSummariesQuery {
  /** Filter by method (optional) */
  method?: string;
  /** Filter by language (optional) */
  language?: string;
  /** Maximum number of summaries to return (optional) */
  limit?: number;
  /** Offset for pagination (optional) */
  offset?: number;
}

File: sdks/csharp/Summarization.cs
using Vectorizer.Models;

namespace Vectorizer;

public partial class VectorizerClient
{
    /// <summary>
    /// Summarize text using various methods
    /// </summary>
    public async Task<SummarizeTextResponse> SummarizeTextAsync(
        SummarizeTextRequest request,
        CancellationToken cancellationToken = default)
    {
        return await RequestAsync<SummarizeTextResponse>(
            "POST", "/summarize/text", request, cancellationToken);
    }

    /// <summary>
    /// Summarize context using various methods
    /// </summary>
    public async Task<SummarizeContextResponse> SummarizeContextAsync(
        SummarizeContextRequest request,
        CancellationToken cancellationToken = default)
    {
        return await RequestAsync<SummarizeContextResponse>(
            "POST", "/summarize/context", request, cancellationToken);
    }

    /// <summary>
    /// Get summary by ID
    /// </summary>
    public async Task<SummarizeTextResponse> GetSummaryAsync(
        string summaryId,
        CancellationToken cancellationToken = default)
    {
        return await RequestAsync<SummarizeTextResponse>(
            "GET", $"/summarize/{Uri.EscapeDataString(summaryId)}", null, cancellationToken);
    }

    /// <summary>
    /// List summaries
    /// </summary>
    public async Task<ListSummariesResponse> ListSummariesAsync(
        ListSummariesQuery? query = null,
        CancellationToken cancellationToken = default)
    {
        var queryString = "";
        if (query != null)
        {
            var paramsList = new List<string>();
            if (query.Limit.HasValue) paramsList.Add($"limit={query.Limit.Value}");
            if (query.Offset.HasValue) paramsList.Add($"offset={query.Offset.Value}");
            if (!string.IsNullOrEmpty(query.Method)) paramsList.Add($"method={Uri.EscapeDataString(query.Method)}");
            if (paramsList.Count > 0) queryString = "?" + string.Join("&", paramsList);
        }

        return await RequestAsync<ListSummariesResponse>(
            "GET", $"/summarize{queryString}", null, cancellationToken);
    }
}

